pkg/config/legacy_converter.go

```diff
@@ -26,8 +26,9 @@ func (c *LegacyConfigConverter) Set(key string, value interface{}) {
 // NewConfigConverter is creating and returning a config converter
 func NewConfigConverter() *LegacyConfigConverter {
 	// Configure Datadog global configuration
-	Datadog = NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	newCfg := NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
```

Comment:
Is removing the side effect safe here ?

Reply:
Yes, we call `SetDatadog` bellow, which update the variable